invenio_users_resources/services/generators.py

```diff
@@ -108,3 +111,31 @@ def query_filter(self, identity=None, **kwargs):
                     return dsl.Q("term", id=need.value)
 
         return []
+
+
+class AdminAction(Generator):
+    """Allows admin-enabled actions."""
+
+    def __init__(self, action):
+        """Constructor."""
+        self.action = action
+        super().__init__()
+
+    def needs(self, **kwargs):
+        """Enabling Needs."""
+        return [self.action]
+
+
+class UserModeration(AdminAction):
```

Comment:
Comment: I would not create this extra class. Simply use ``AdminAction(user_management_action)``

If needed in permissions.py do:

```
can_manage = AdminAction(user_management_action)
can_read = [can_manage,...]
```